docs/gallery/index.html

```diff
@@ -23,9 +41,120 @@
   </header>
-
+<div id="gallery"></div>
 </body>
+<script src="carousel.js"></script>
+<script src="icon.js"></script>
+<script>
+  function groupArray(arr, size = 3) {
+    const groups = []
+    for (let i = 0; i < arr.length; i += size) {
+      groups.push(arr.slice(i, i + size))
+    }
+    return groups
+  }
+  async function mainIfHome(gallery) {
+    console.log(1)
+   
+    document.getElementById('headline').innerText = `Gallery of the ${gallery.length} rices`
+  // console.log(groupArray(gallery), 0)
+const grouped = groupArray(gallery)
+let index = 0;
+for(const group of grouped) {
+  const row=document.createElement('div')
+  row.classList.add('row')
+  for(const cold of group) {
+    const col=document.createElement('a')
+    col.classList.add('card')
+    col.classList.add('col')
+    col.href = `#${index}`
+    const img = document.createElement('img')
+    // lazy load
+    img.setAttribute('loading', 'lazy')
+    img.src = cold.images[0]
+    img.alt = cold.name
+    img.style.width = "90%"
+    img.style.height = "90%"
+    img.style.borderRadius = "10px"
+    const h1 = document.createElement('h1')
+    // h1.append(setupIcon(cold.distro))
+    h1.innerText = cold.name
+    h1.style.paddingBottom = "10px"
+    col.appendChild(h1)
+    col.appendChild(img)
+    row.appendChild(col)
+  index++
+  }
+
+  document.getElementById('gallery').appendChild(row)
+  }
+}
+async function mainIfViewIndex(gallery) {
+  const index = parseInt(window.location.hash.slice(1))
+  const data =gallery[index]
+  if(!data) window.location.hash = ''
+  console.log(data)
+  const carousel = document.createElement('div')
+  carousel.classList.add('carousel-container')
+  const imgCarousel = document.createElement('div')
+  imgCarousel.classList.add('carousel')
+  for(const image of data.images) {
+    const img = document.createElement('img')
+    img.src = image
+    img.alt = data.name
+    img.style.width = "100%"
+    img.style.height = "auto"
+    imgCarousel.appendChild(img)
+  }
+  carousel.appendChild(imgCarousel)
+  const next = document.createElement('button')
+  next.classList.add('next')
+  next.innerText = 'Next Image'
+  const prev = document.createElement('button')
+  prev.classList.add('prev')
+  prev.innerText = 'Prev Image'
+  if(data.images.length > 1){
+    carousel.appendChild(prev)
+    carousel.appendChild(next)
+  }
+  const titleThing = document.createElement('h1')
+  titleThing.innerText = data.name
+  const h1 = titleThing
+  h1.style.textAlign = "center"
+    // h1.style.margin = "auto"
+    const gitLink = document.createElement('a')
+    gitLink.href = data.git
+    gitLink.innerText = "Dotfiles"
+    gitLink.style.textDecoration = "none"
+    gitLink.target = "_blank"
+    h1.style.verticalAlign = "center"
+    h1.appendChild(document.createTextNode(" - "))
+    h1.appendChild(gitLink)
+    h1.appendChild(document.createTextNode(" "))
+    h1.appendChild(setupIcon(data.distro))

```

Comment:
Still feels a bit off-centered. I ended up fixing this on my local copy with making it a flexbox container:

- Making the container a `div`
- Aligning horizontally with `justify-content: center;`
- Aligning vertically with `align-items: center;`
- Adding the gap between the items with `gap` (I used `var(--spacing-2)` which is 8px).
- Creating the `<h1>` effect with `font-size: var(--font-5); font-weight: var(--font-weight-bold);`

```suggestion
  const titleThing = document.createElement('div')
  titleThing.innerText = data.name
  const div = titleThing
  div.style.display = "flex"
  div.style.justifyContent = "center"
  div.style.alignItems = "center"
  div.style.gap = "var(--spacing-2)"
  div.style.fontSize = "var(--font-5)"
  div.style.fontWeight = "var(--font-weight-bold)"
  
  const gitLink = document.createElement('a')
  gitLink.href = data.git
  gitLink.innerText = "Dotfiles"
  gitLink.style.textDecoration = "none"
  gitLink.target = "_blank"
  
    div.appendChild(document.createTextNode(" - "))
    div.appendChild(gitLink)
    div.appendChild(setupIcon(data.distro))
```